diff --git a/luci-app-homeproxy/Makefile b/luci-app-homeproxy/Makefile
index 8af7460f..43abe034 100644
--- a/luci-app-homeproxy/Makefile
+++ b/luci-app-homeproxy/Makefile
@@ -5,11 +5,6 @@
 include $(TOPDIR)/rules.mk
 
 LUCI_TITLE:=The modern ImmortalWrt proxy platform for ARM64/AMD64
-LUCI_DEPENDS:= \
-	+sing-box \
-	+chinadns-ng \
-	+firewall4 \
-	+kmod-nft-tproxy
 
 define Package/luci-app-homeproxy/conffiles
 /etc/config/homeproxy
