diff --git a/luci-app-advancedplus/luasrc/model/cbi/advancedplus/kucatset.lua b/luci-app-advancedplus/luasrc/model/cbi/advancedplus/kucatset.lua
index 98676194..67e12f51 100644
--- a/luci-app-advancedplus/luasrc/model/cbi/advancedplus/kucatset.lua
+++ b/luci-app-advancedplus/luasrc/model/cbi/advancedplus/kucatset.lua
@@ -43,6 +43,7 @@ e:value('1', translate('Auto download Iciba wallpaper'))
 e:value('2', translate('Auto download unsplash wallpaper'))
 e:value('3', translate('Auto download Bing wallpaper'))
 e:value('4', translate('Auto download Bird 4K wallpaper'))
+e:value('6', translate('Auto download ACG wallpaper'))
 e.default = '0'
 e.rmempty = false
 
diff --git a/luci-theme-kucat/root/usr/libexec/kucat/kucat_wallpaper b/luci-theme-kucat/root/usr/libexec/kucat/kucat_wallpaper
index aecfd072..7b4c6db3 100644
--- a/luci-theme-kucat/root/usr/libexec/kucat/kucat_wallpaper
+++ b/luci-theme-kucat/root/usr/libexec/kucat/kucat_wallpaper
@@ -56,6 +56,9 @@ curl_bgurl() {
             "https://wallhaven.cc/api/v1/search?resolutions=1920x1080&sorting=random" |
             jsonfilter -qe '@.data[0].path'
 		;;
+	6)
+		echo "https://api.vvhan.com/api/wallpaper/acg"
+		;;

 	esac
 }
@@ -98,4 +101,4 @@ else
 fi
 }

-check_url
\ No newline at end of file
+check_url
