diff --git a/fibocom_QMI_WWAN/src/qmi_wwan_f.c b/fibocom_QMI_WWAN/src/qmi_wwan_f.c
index 8995e832..c8b3f192 100755
--- a/fibocom_QMI_WWAN/src/qmi_wwan_f.c
+++ b/fibocom_QMI_WWAN/src/qmi_wwan_f.c
@@ -2096,8 +2096,9 @@ static int qmi_wwan_reset_resume(struct usb_interface *intf)
 
 static int rmnet_usb_bind(struct usbnet *dev, struct usb_interface *intf)
 {
+    int status;
     dev_err(&intf->dev, "rmnet_usb_bind\n");
-    int status = qmi_wwan_bind(dev, intf);
+    status = qmi_wwan_bind(dev, intf);
 
     if (!status) {
         struct qmi_wwan_state *info = (void *)&dev->data;
